File: C/hard/sub in string.c
#include <stdio.h>
#include <string.h>

int main() {
    char str[100], sub[100];
    printf("Enter a string: ");
    scanf("%s", str);
    printf("Enter a subsequence: ");
    scanf("%s", sub);
    if (strstr(str, sub) != NULL) {
        printf("The subsequence is present in the string.\n");
    } else {
        printf("The subsequence is not present in the string.\n");
    }
    return 0;
}

File: JAVA/medium/sub in string.java
import java.util.Scanner;

public class Main {
    public static void main(String[] args) {
        Scanner scanner = new Scanner(System.in);
        System.out.print("Enter a string: ");
        String str = scanner.nextLine();
        System.out.print("Enter a subsequence: ");
        String sub = scanner.nextLine();
        if (str.contains(sub)) {
            System.out.println("The subsequence is present in the string.");
        } else {
            System.out.println("The subsequence is not present in the string.");
        }
    }
}
